src/inmanta/agent/forking_executor.py

```diff
@@ -681,3 +708,14 @@ async def stop_for_agent(self, agent_name: str) -> list[MPExecutor]:
         children = [self.executor_map[child_id] for child_id in children_ids]
         await asyncio.gather(*(child.stop() for child in children))
         return children
+
+    async def cleanup_inactive_executors(self) -> None:
+        now = datetime.datetime.now().astimezone()
+
+        for _executor in self.executor_map.values():
```

Comment:
Instead of calling this on a schedule and to go over all executors every two seconds, how about the following?

1. in this method, stop all timed out executors and save `next_expiry = min(retention_time - (now - last_used))` (pseudo-code)
2. at the end of this method, reschedule itself on the IO loop in `next_expiry` seconds.
3. When the setting is changed, we may need to take some action to reschedule.

We would gain the following advantages:
a. More responsive: clean up after 60 seconds rather than after 61.9 seconds. Not too important IMO
b. Safer: no risk of being in this method twice.
c. More to the point: on low-load times we may end up just calling this method every 60 seconds.

They're not big gains, but I'd say it's an easy win, and it feels like a cleaner approach to me.

@wouterdb FYI